packages/svelte2tsx/src/svelte2tsx.ts

```diff
@@ -655,10 +655,8 @@ function processInstanceScriptContent(str: MagicString, script: Node): InstanceS
 
         str.prependLeft(start, '__sveltets_invalidate(() => ');
         str.appendRight(end, ')');
-
-        if (!semiRegex.test(htmlx.substring(end))) {
```

Comment:
This is added by me to prevent this
```
$: [ first ]  = a
['a', 'b'].forEach(item => something(item))
```
because the invalidate function call the last character might be a ')' so the semi might sometimes be needed

Reply:
Nevermind I am just having examples that don't works 😅

Reply:
Oh ok, I did not think of this situation. The problem with that added semicolon is that I did not find a way to to make it work in combination with the generated output of `$: ({x} = y)`.

Reply:
Nevermind the example doesn't work. it requires a semi before transformation anyways. I must have set that semi for other reasons but I forgot 😂